File: src/ddd/simple/service/major_module/MajorService.java
package ddd.simple.service.major_module;

import ddd.base.persistence.EntitySet;
import ddd.simple.service.base.BaseServiceInterface;
import ddd.simple.entity.major_module.Major;

public interface MajorService extends BaseServiceInterface {
	public Major saveMajor(Major major) ;
	
	public int deleteMajor(Long majorId) ;
	
	public Major updateMajor(Major major) ;
	
	public Major findMajorById(Long majorId) ;
	
	public EntitySet<Major> findAllMajor() ;
 
}

File: WebContent/ddd/simple/view/major_module/js/major/MajorService.js

define(["app"], function (app) {
	app.factory("MajorService", function($resource) {
		
		var majorService = $resource("../DD/Major/:action", {});
		majorService.saveMajor=function(major,sucesscb,errorcb)	{
			majorService.save({action:"saveMajor"},{major:major},sucesscb,errorcb);
		};
		
		majorService.deleteMajor=function(majorId,sucesscb,errorcb)	{
			majorService.save({action:"deleteMajor"},{majorId:majorId},sucesscb,errorcb);
		};
		
		majorService.updateMajor=function(major,sucesscb,errorcb) {
			majorService.save({action:"updateMajor"},{major:major},sucesscb,errorcb);
		};
		 
		majorService.findMajorById=function(majorId,sucesscb,errorcb) {
			majorService.get({action:"findMajorById",majorId:majorId},sucesscb,errorcb);
		};
		
		majorService.findAllMajor=function(sucesscb,errorcb) {
			majorService.query({action:"findAllMajor"},sucesscb,errorcb);
		}
		return majorService;
})});
